### PR TITLE
Remove add command

### PR DESCRIPTION
**Pull request by anamnavi**
## PR Summary

For internal compliance reasons we will not be downloading the powershell package from online (through ADD or curl) directly into our images.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
  - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `Dockerfile`, `.sh`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
  - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.

### CODE REVIEW

File: tools/buildHelper/buildHelper.psm1 [hunk 1]
```diff
@@ -721,13 +721,80 @@ function Get-TestParams
...
+                # $pwshInstallerFileUri = "https://github.com/PowerShell/PowerShell/releases/download/v$($psversion)/PowerShell-$($psversion)-win-x64.zip"                
+                $pwshInstallerFileName = "PowerShell-$($psversion)-win-x64.zip"
+            }
+            "ubi"
+            {
+                # $pwshInstallerFileUri = "https://github.com/PowerShell/PowerShell/releases/download/v$($psversion)/powershell-preview-$($psversion)-1.rh.x86_64.rpm"
+                $pwshInstallerFileName = "powershell-preview-$($psversion)-1.rh.x86_64.rpm"
+            }
+        }
+    }
```
Comment (TravisEz13) [inline tools/buildHelper/buildHelper.psm1:779]: We already calculate the package name on lines 808 and 832 (in your version).
Your version hard codes the package format which is incorrect.  Meta already has this information, so we don't have to change the code when the package format changes.

Also, Mariner uses tar.gz for stable and rpm for preview (currently.)

File: release/unstable/clearlinux/docker/Dockerfile [hunk 2]
```diff
@@ -13,7 +13,7 @@ ARG PS_PACKAGE_URL=https://github.com/PowerShell/PowerShell/releases/download/v$
 ARG PS_INSTALL_VERSION=7
 
 # Download the PowerShell Core Linux tar.gz and save it
-ADD ${PS_PACKAGE_URL} /tmp/powershell-linux.tar.gz
+RUN curl -sSL ${PS_PACKAGE_URL} -o /tmp/powershell-linux.tar.gz
```
Comment (TravisEz13) [inline release/unstable/clearlinux/docker/Dockerfile:16]: We can' use curl either :(

Comment (anamnavi) [reply release/unstable/clearlinux/docker/Dockerfile:16]: need to remove ADD and curl from unstable channel images.

File: release/unstable/blackarch/docker/Dockerfile [hunk 3]
```diff
@@ -52,7 +52,7 @@ COPY --from=installer-env ["/tmp/gss-ntlmssp/gss-ntlmssp.pkg.tar.zst", "/tmp/gss
 ARG BlackArch_Strap_URL=https://blackarch.org/strap.sh
 
 # Download BlackArch Linux strap shell script
-ADD ${BlackArch_Strap_URL} /tmp/strap.sh
+RUN curl -sSL ${BlackArch_Strap_URL} -o /tmp/strap.sh
```
Comment (TravisEz13) [inline release/unstable/blackarch/docker/Dockerfile:55]: let's revert this change because it doesn't have to do with our package.

File: release/7-3/ubuntu22.04/docker/Dockerfile [hunk 4]
```diff
@@ -5,35 +5,9 @@
...
-    && curl -sSL ${PS_PACKAGE_URL} -o /tmp/powershell.deb
-
-# Install the deb file in this image and make powershell available
-FROM ubuntu:22.04 AS final-image
-
-# # Define args needed to add the package
-ARG PS_VERSION=7.3.0-preview.4
-ARG PS_PACKAGE=powershell-preview_${PS_VERSION}-1.deb_amd64.deb
-ARG PS_PACKAGE_URL=https://github.com/PowerShell/PowerShell/releases/download/v${PS_VERSION}/${PS_PACKAGE}
-ARG PS_INSTALL_VERSION=7-preview
```
Comment (TravisEz13) [inline release/7-3/ubuntu22.04/docker/Dockerfile:36]: This is used on line 42

File: release/7-3/debian11/docker/Dockerfile [hunk 5]
```diff
@@ -6,36 +6,12 @@
...
-FROM debian:bullseye-slim
-
-ARG PS_VERSION=6.2.0-preview.3
-ARG PS_INSTALL_VERSION=7-preview
-
-# Copy only the files we need from the previous stage
-COPY --from=installer-env ["/opt/microsoft/powershell", "/opt/microsoft/powershell"]
-
-# Define Args and Env needed to create links
-ARG PS_INSTALL_VERSION=7-preview
```
Comment (TravisEz13) [inline release/7-3/debian11/docker/Dockerfile:36]: Used on the next line.

Check other images

File: tools/buildHelper/buildHelper.psm1 (hunk 1 above)
Comment (anamnavi) [reply tools/buildHelper/buildHelper.psm1:779]: ah good point, I'll pull the package name from the meta. Then I also don't need to populate that based on OS, it can take what we specify in the meta

Comment (TravisEz13) [reply tools/buildHelper/buildHelper.psm1:779]: Why do we need a new property?

File: release/unstable/clearlinux/docker/Dockerfile (hunk 2 above)
Comment (TravisEz13) [reply release/unstable/clearlinux/docker/Dockerfile:16]: We can ignore unstable.

**Comment by RDIL** [comment]
What exactly is the reasoning behind this change?

**Comment by anamnavi** [comment]
@RDIL  for internal compliance reasons we don't want to be downloading powershell package from online (through ADD or curl) directly into our images

**Comment by RDIL** [comment]
What is the plan for getting the package into the images then?

**Comment by anamnavi** [comment]
@RDIL users will run the build script first, which will download the powershell installer file, then in the image we copy that installer file in and install the package.

File: tools/buildHelper/buildHelper.psm1 [hunk 6]
```diff
@@ -721,13 +733,34 @@ function Get-TestParams
...
 
+    # get tmp cache location call method
+    $tmpCacheFolder = Get-CacheFolder
+    $cachedPwshFilePath = Join-Path -Path $tmpCacheFolder -ChildPath $allmeta.meta.PackageFormat
+
+    # check if file already exists in cache
+    if (!(Test-Path $cachedPwshFilePath))
+    {
+        # download the powershell installer file
+        $pwshSourceInstallerFile = "https://github.com/PowerShell/PowerShell/releases/download/v$($psversion)/$($allmeta.meta.PackageFormat)"
```
Comment (anamnavi) [inline tools/buildHelper/buildHelper.psm1:744]: use code to get URL (may be a function, or refactor into function

File: build.ps1 [hunk 7]
```diff
@@ -156,8 +156,14 @@ param(
...
     [string]
     [ValidateSet('All','Linux','Windows')]
-    $OsFilter
+    $OsFilter,
 
+    [Parameter(ParameterSetName="localBuildByName")]
+    [Parameter(ParameterSetName="localBuildAll")]
+    [Parameter(Mandatory, ParameterSetName="TestByName")]
+    [Parameter(Mandatory, ParameterSetName="TestAll")]
+    [switch]
```
Comment (anamnavi) [inline build.ps1:165]: remove parameter sets, let it be in all p sets.

**Review by Ray0light — APPROVED** [review]

**Review by Ray0light — APPROVED** [review]

**Review by Ray0light — APPROVED** [review]

**Review by Ray0light — APPROVED** [review]

**Review by Ray0light — APPROVED** [review]
